src/sync/installation.ts

```diff
@@ -207,7 +213,12 @@ const doProcessInstallation = async (data: BackfillMessagePayload, sentry: Hub,
 	const gitHubProduct = getCloudOrServerFromGitHubAppId(subscription.gitHubAppId);
 
 	if (!nextTask) {
-		await subscription.update({ syncStatus: "COMPLETE" });
+
+		const newBackFillSinceDate: Date | undefined = await booleanFlag(BooleanFlags.USE_BACKFILL_ALGORITHM_INCREMENTAL, jiraHost)
```

Comment:
Can we wrap this into a function?

Already so many things going on in this file would be nice to break into pieces, looks like your doind the same update on line 122 there so could re-use too